native/src/main/java/io/ballerina/lib/data/yaml/emitter/Emitter.java

```diff
@@ -0,0 +1,317 @@
+
+        public List<BString> getDocument() {
+            return getDocument(false);
+        }
+
+        public List<BString> getDocument(boolean isStream) {
+            List<BString> output = new ArrayList<>(document.stream().toList());
+            if (isStream && document.size() > 0) {
+                output.add(0, START_OF_YAML_DOCUMENT);
+                lastBareDoc = true;
+            }
+
+            document = new ArrayList<>();
+            return output;
+        }
+    }
+
+    public static List<BString> emit(EmitterState state, boolean isStream) {
+        if (isStream) {
+            List<BString> output = new ArrayList<>();
+            boolean isFirstEvent = true;
+            while (!state.events.isEmpty()) {
+                write(state);
+                output.addAll(state.getDocument(!isFirstEvent));
+                isFirstEvent = false;
+            }
+            return output;
+        }
+        write(state);
+        return state.getDocument();
+    }
+
+    private static void write(EmitterState state) {
+        YamlEvent event = getEvent(state);
+
+        if (event.getKind() == YamlEvent.EventKind.START_EVENT) {
+            YamlEvent.StartEvent startEvent = ((YamlEvent.StartEvent) event);
+            Types.Collection startType = startEvent.getStartType();
+            if (startType == Types.Collection.SEQUENCE) {
+                if (startEvent.isFlowStyle()) {
+                    state.addLine(writeFlowSequence(state, event.getTag()));
+                } else {
+                    writeBlockSequence(state, "", event.getTag());
+                }
+                return;
+            }
+            if (startType == Types.Collection.MAPPING) {
+                if (startEvent.isFlowStyle()) {
+                    state.addLine(writeFlowMapping(state, event.getTag()));
+                } else {
+                    writeBlockMapping(state, "");
+                }
+                return;
+            }
+        }
+
+        if (event.getKind() == YamlEvent.EventKind.SCALAR_EVENT) {
+            YamlEvent.ScalarEvent scalarEvent = (YamlEvent.ScalarEvent) event;
+            state.addLine(writeNode(state, scalarEvent.getValue(), event.getTag()));
+        }
+    }
+
+    private static String writeNode(EmitterState state, String value, String tag) {
+        return writeNode(state, value, tag, false);
+    }
+
+    private static String writeNode(EmitterState state, String value, String tag, boolean tagAsSuffix) {
+        if (tag == null) {
+            return value;
+        }
+
+        if (tag.startsWith(DEFAULT_GLOBAL_TAG_HANDLE)) {
+            return state.canonical ? appendTagToValue(tagAsSuffix, "!!" +
+            tag.substring(DEFAULT_GLOBAL_TAG_HANDLE.length()), value) : value;
+        }
+
+        return "";
+    }
+
+    private static void writeBlockMapping(EmitterState state, String whitespace) {
+        YamlEvent event = getEvent(state);
+        String line;
+
+        while (true) {
```

Comment:
```suggestion
        while (event.getKind() != YamlEvent.EventKind.END_EVENT) {
```

Reply:
No we cannot do this because we have to assign line to empty.